src/aind_metadata_extractor/smartspim/extractor.py

```diff
@@ -0,0 +1,146 @@
+"""SmartSPIM extractor module"""
+
+from datetime import datetime
+import os
+import re
+from typing import Optional
+from pathlib import Path
+
+import requests
+
+from aind_metadata_extractor.smartspim.job_settings import JobSettings
+from aind_metadata_extractor.models.smartspim import SmartspimModel, FileMetadataModel, SlimsMetadataModel
+from aind_metadata_extractor.smartspim.utils import get_excitation_emission_waves, get_session_end, read_json_as_dict
+
+REGEX_DATE = r"(20[0-9]{2})-([0-9]{2})-([0-9]{2})_([0-9]{2})-" r"([0-9]{2})-([0-9]{2})"
+REGEX_MOUSE_ID = r"([0-9]{6})"
+
+
+class SmartspimExtractor:
+    """Extractor for SmartSPIM metadata from microscope files and SLIMS."""
+
+    def __init__(self, job_settings: dict):
```

Comment:
it'd be nice if it was an option to pass in a JobSettings object. It would look something like this: `job_settings: Union[Dict, JobSettings]`

Reply:
or just `JobSettings` would be better.